.github/workflows/apply-new-account-baseline.yml

```diff
@@ -63,3 +68,13 @@ jobs:
           echo "TERRAGRUNT_COMMAND: $TERRAGRUNT_COMMAND"
           echo "MANAGEMENT_ACCOUNT: $MANAGEMENT_ACCOUNT"
           echo "CHILD_ACCOUNT: $CHILD_ACCOUNT"
+
+          if [[ -z ${PRESIGNED_CALLER_IDENTITY:-} ]]; then
+            echo "PRESIGNED_CALLER_IDENTITY is not set"
+            exit 0
```

Comment:
Should this be `exit 1`? This applies here and to the similar checks in the other workflows below.

Reply:
@brikis98 

It's kind of confusing, but I'm testing both true and false variants of the `presign_token` inputs by dispatching a `workflow_dispatch` from `gruntwork-io/pipelines-dispatch` to `gruntwork-io/pipelines-dispatch`, simulating what will happen in `infrastructure-pipelines` in the relevant workflows in either scenario:
https://github.com/gruntwork-io/pipelines-dispatch/blob/enhancement/adding-pipelines-auth-testing/.github/workflows/test-caller.yml

In the variant where we don't set `presign_token`, we expect for the input to be empty, in which case, we can safely ignore in the input.

In the variant where we do set `presign_token` we expect the value of `fake token` to appear due to the fact that we use the pretend pipelines cli [here](https://github.com/gruntwork-io/pipelines-dispatch/pull/27/files#diff-1c8d1f3af9947e44e4ff901b975af0c9fdd850fb6d72f2ed921ad1864051cfaaR46-R51). This means that when the action attempts to presign a token, it'll just run `echo fake token` instead of actually presigning a token.

Reply:
Ah. Perhaps those `echo` statements should make that clearer? E.g., `echo "PRESIGNED_CALLER_IDENTITY is not set. Assuming this is our test workflow where we set this value to empty, which means..."`

Also, if we are using a fake `pipelines` CLI that doesn't sign anything, are we really testing the workflow?

Reply:
I think so, as whether or not the pipelines CLI works, this action will still be doing what it should be doing, which is sending the appropriate payload to the `infra-pipelines` repo determined by invoking it, and setting the stdout of the command to the appropriate workflow input when it performs the dispatch.

Reply:
I'll make those `echo`s clearer.

Reply:
> I think so, as whether or not the pipelines CLI works, this action will still be doing what it should be doing, which is sending the appropriate payload to the `infra-pipelines` repo determined by invoking it, and setting the stdout of the command to the appropriate workflow input when it performs the dispatch.

Again, I apologize if I'm missing context, but the title of the PR is "testing pipelines **auth**" (emphasis mine), so if we're not signing anything, how is this testing auth?

Reply:
I think it's more likely that I'm assuming what I'm doing is sufficient, when I'm missing something.

It's testing that the `pipelines-dispatch` action correctly invokes the `pipelines` CLI, and appends the stdout of that operation to the JSON payload sent to the correct workflow.

If we wanted to test that this worked fully without mocking anything, we would have to maintain a secret for the pipelines CLI repo and secrets for AWS authentication in order for the action to actually install the pipelines CLI and presign a `GetCallerIdentity` request that would pass verification on the other end.

I figured that wasn't necessary because the code in this repo is not really concerned with the validity of the signing procedure or whether or not the pipelines CLI runs correctly when installed. It just needs to correctly run the CLI when input parameters indicate that it should, and that the JSON payload constructed to be delivered to the relevant workflows has that `presigned_caller_identity` field added (and my assumption is that the value doesn't really matter for verification that the code in this action works).

If it's very important to us that we run the real pipelines CLI, I just need to work out a good way to maintain secrets on this repo that can install that pipelines CLI. I'm hesitant to do this because it's a public repo, and I don't want to risk exposure of something like that.

If it's not important to us that we use real AWS credentials, I can mock those with values like `abc` and `123`, as the signing procedure still works normally in the AWS SDK regardless of whether the credentials are real.

Reply:
Roger, thanks for the explanation. Perhaps add that as a comment to the tests so future maintainers know what we are trying to test here and what we aren't?

Reply:
Comment added.